lib/compress/zstd_fast.c

```diff
@@ -587,11 +587,10 @@ static size_t ZSTD_compressBlock_fast_extDict_generic(
     U32* const hashTable = ms->hashTable;
     U32 const hlog = cParams->hashLog;
     /* support stepSize of 0 */
-    U32 const stepSize = cParams->targetLength + !(cParams->targetLength);
+    size_t const stepSize = hasStep ? (cParams->targetLength + !(cParams->targetLength) + 1) : 2;
```

Comment:
Hardcoding hasStep by itself improves clang12 compression speed by 0.2% and doesn't affect gcc11. However, combining hasStep hardcoding with repcode-safety hardcoding improves gcc11 compression speed by 0.4% and clang12 by 0.9%. This is significant enough to justify a variant IMO. Here's the code I tested: https://gist.github.com/embg/e79fa67ab63db1e9634ce63e1af0aebe

The same idea can also be transferred to noDict in a separate PR, since it also has repcode safety checks which are unnecessary 99%+ of the time. The number of variants doesn't need to be increased since the "common case" can be folded into a single variant `(hasStep = 0 & repcodeOffsetsAreSafe)`.
```
/home/embg/zstd/largeNbDicts_2c0540_clang12
110.0
109.9
109.9
/home/embg/zstd/largeNbDicts_2c0540_gcc11
106.3
106.3
106.3
/home/embg/zstd/largeNbDicts_a438e8e_clang12
110.9
110.9
111.0
/home/embg/zstd/largeNbDicts_a438e8e_gcc11
106.7
106.7
106.8
```